conanfile.txt

```diff
@@ -0,0 +1,16 @@
+# Docs at https://docs.conan.io/en/latest/reference/conanfile_txt.html
+
+[requires]
+catch2/2.13.7
+docopt.cpp/0.6.2
+fmt/8.0.1
+spdlog/1.9.2
+# imgui-sfml/2.1@bincrafters/stable
+# sdl2/2.0.10@bincrafters/stable
+
+[generators]
+cmake_find_package
```

Comment:
Are all those generators necessary? I would prefer the conan to be less intrusive to the CMakeList.txt files. There is the option to generate a toolchain file as described [here]( https://docs.conan.io/en/1.32/creating_packages/toolchains/cmake.html) and then pass it (if you want) to the CMake command. I am not familiar to the vcpkg but as I can see [here](https://vcpkg.readthedocs.io/en/latest/examples/installing-and-using-packages/) it also produce a toolchain .cmake file that can be passed as an `CMAKE_TOOLCHAIN_FILE` argument. This way the user can chose between conan and vcpkg at will, without the need for a package specific logic inside the CMakeLists.txt

Reply:
Also I do not see any benefits of converting conanfile.py to conanfile.txt since all developer machines have python installed already and the capabilities of the .txt file are way less powerful than the .py

Reply:
See https://github.com/cpp-best-practices/cpp_starter_project/pull/169#discussion_r715242962 and following comments. The .txt is declarative and does not require a developer to understand Python to debug it.

Reply:
These generators can be removed if are not needed. There is no noticeable overhead in supporting all of them, so I think there is no harm in specifying all of them. 

vcpkg is already supported using `run_vcpkg` function.